app/master/build.py

```diff
@@ -309,6 +288,41 @@ def _on_enter_error_state(self, event):
         self._error_message = getattr(event, 'error_msg', default_error_msg)
         self._logger.warning('Build {} failed: {}', self.build_id(), self._error_message)
 
+    def _on_enter_preparing_state(self, event):
+        """
+        Prepare the build by atomization and subjobs creation.
+        This method is triggered by a state machine transition to the PREPARING state.
+        :param event: The Fysom event object
+        """
+        self._logger.info('Fetching project for build {}.', self._build_id)
+        self.project_type.fetch_project()
+        self._logger.info('Successfully fetched project for build {}.', self._build_id)
+
+        job_config = self.project_type.job_config()
+        if job_config is None:
+            raise RuntimeError('Build failed while trying to parse clusterrunner.yaml.')
+
+        subjob_calculator = getattr(event, 'subjob_calculator', None)
```

Comment:
Sorry, I thought that the build callback allowed you to specify additional kwargs to the function. I didn't realize that it sets a member variable within the Event object instead.

Instead of using `getattr`, how about just instantiating a new instance of SubjobCalculator instead.

```
subjob_calculator = SubjobCalculator()
vs
subjob_calculator = getattr(event, 'subjob_calculator', None)
```